### PR TITLE
fix: update wasm functions to accept `WasmModuleHandler`

### PR DESCRIPTION
**Pull request by qartik**
# Description

The `add_wasm` and `add_wasm_to_reg` functions should now accept `WasmModuleHandler` which is a super class of `WasmFileHandler`.

# Related issues

Follow up to #1503

# Checklist

- [x] I have performed a self-review of my code.
- [ ] I have commented hard-to-understand parts of my code.
- [ ] I have made corresponding changes to the public API documentation.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [ ] I have updated the changelog with any user-facing changes.

### CODE REVIEW

**Comment by qartik** [comment]
Hi @cqc-melf, unsure if the errors reported on the CI are because of the changes in the PR. Could you help take a look?

**Comment by cqc-alec** [comment]
The `.pyi` file needs to be generated using the `generate_stubs` script (I can do that and push the change...).

**Comment by cqc-alec** [comment]
I don't know why the coverage check is failing. I think there is some random variation in it, but I haven't seen it fail for a long time. Will try rerunning jobs just in case it's a random outlier.

**Comment by cqc-alec** [comment]
OK, I don't know why the coverage has gone down, maybe the previous coverage metric was abnormally high. Happy to ignore that failure, if @cqc-melf is otherwise happy with the changes.

**Comment by qartik** [comment]
> I don't know why the coverage has gone down

it could possibly be because all the existing tests are still passing `WasmFileHandler`, while the function accepts `WasmModuleHandler`, let me add a test for the latter as well.

**Review by cqc-melf — APPROVED** [review]
From my perspective this looks good now :)

**Comment by cqc-melf** [comment]
@cqc-alec Could you merge this maybe?

File: pytket/pytket/circuit/__init__.py [hunk 1]
```diff
@@ -72,7 +72,7 @@ def overload_add_wasm(
 ) -> Circuit:
     """Add a classical function call from a wasm file to the circuit.
     \n\n:param funcname: name of the function that is called
-    \n:param filehandler: wasm file handler to identify the wasm file
+    \n:param filehandler: wasm file or module handler to identify the wasm module
```
Comment (cqc-alec) [inline pytket/pytket/circuit/__init__.py:75]: Shouldn't we change the type annotation to a `Union` if both types are accepted (not sure why mypy didn't complain)?

File: pytket/pytket/circuit/__init__.py [hunk 2]
```diff
@@ -113,15 +113,15 @@ def overload_add_wasm(
...
+    filehandler: wasm.WasmModuleHandler,
     list_i: Sequence[BitRegister],
     list_o: Sequence[BitRegister],
     args_wasm: Optional[Sequence[int]] = None,
     **kwargs: Any,
 ) -> Circuit:
     """Add a classical function call from a wasm file to the circuit.
     \n\n:param funcname: name of the function that is called
-    \n:param filehandler: wasm file handler to identify the wasm file
+    \n:param filehandler: wasm file or module handler to identify the wasm module
```
Comment (cqc-alec) [inline pytket/pytket/circuit/__init__.py:124]: As above.

File: pytket/pytket/circuit/__init__.py (hunk 1 above)
Comment (qartik) [reply pytket/pytket/circuit/__init__.py:75]: Like I mentioned in the OP, `WasmFileHandler` is a subclass of `WasmModuleHandler`. If we are accepting the superclass, any of its subclasses is acceptable.

Comment (qartik) [reply pytket/pytket/circuit/__init__.py:75]: But if you prefer the Union for better documentation from types, I am happy to make that change, let me know.

**Review by cqc-alec — APPROVED** [review]

**Pull request merged**